File: Task 3/Java/app/src/main/java/dev/ogabek/java/network/services/EmployeeService.java
package dev.ogabek.java.network.services;

import java.util.List;

import dev.ogabek.java.model.Employee;
import retrofit2.Call;
import retrofit2.http.*;

public interface EmployeeService {

    @GET("m6-l7-t3")
    Call<List<Employee>> get();

    @PUT("m5-l7-t3/{id}")
    Call<Employee> put(@Path("id") int id, @Body Employee employee);

    @POST("m5-l7-t3")
    Call<Employee> post(@Body Employee employee);

    @DELETE("m5-l7-t3/{id}")
    Call<Employee> delete(@Path("id") int id);

}

File: Task 3/Kotlin/app/src/main/java/dev/ogabek/kotlin/network/EmployeeService.kt
package dev.ogabek.kotlin.network

import dev.ogabek.kotlin.model.Employee
import retrofit2.Call
import retrofit2.http.*

interface EmployeeService {

    @GET("m6-l7-t3")
    fun get(): Call<List<Employee>>

    @POST("m6-l7-t3")
    fun post(@Body employee: Employee): Call<Employee>

    @PUT("m6-l7-t3/{id}")
    fun put(@Path("id") id: Int, @Body employee: Employee): Call<Employee>

    @DELETE("m6-l7-t3/{id}")
    fun delete(@Path("id") id: Int): Call<Employee>

}